### ISSUE TITLE
[Accessibility]: Branches page: 320px usability issues

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35462]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34424

### Problem description

<img width="396" alt="image" src="https://user-images.githubusercontent.com/9516420/168571870-03875439-4c9b-43ec-bd7f-91cbe1951441.png">

**Issue 1**
Branch text is completely cut off and barely readable.

**Issue 2**
Repo header clashes with "three vertical dots" symbol. Hard to click it on a screen like this.

### Expected behavior

**Issue 1**
We should use `flex-direction: column` to make the branch content appear vertically on small screens

**Issue 2**
We should break out the repository header onto two lines on small screens: One line for the breadcrumbs, one line for all other actions: Related issue https://github.com/sourcegraph/sourcegraph/issues/35239

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
